pulp_2to3_migration/app/plugin/content.py

```diff
@@ -407,61 +408,66 @@ def get_remote_by_importer_id(importer_id):
+                                await self.put(dc)
+
+                            if not at_least_one_lce_migrated:
+                                _logger.warn(
+                                    _('On_demand content cannot be migrated without a remote '
+                                      'pulp2 unit_id: {}'.format(pulp2content.pulp2_id)
+                                      )
+                                )
+                            future_relations.update({'lces': list(pulp2lazycatalog)})
+                        else:
                             relative_path = (
                                 pulp_2to3_detail_content.relative_path_for_content_artifact
                             )
                             da = DeclarativeArtifact(
                                 artifact=artifact,
-                                url=lce.pulp2_url,
+                                url=NOT_USED,
                                 relative_path=relative_path,
-                                remote=remote,
-                                deferred_download=deferred_download)
-                            lce.is_migrated = True
-                            at_least_one_lce_migrated = True
+                                remote=None,
+                                deferred_download=False)
                             dc = DeclarativeContent(content=pulp3content, d_artifacts=[da])
                             dc.extra_data = future_relations
                             await self.put(dc)
 
-                        if not at_least_one_lce_migrated:
-                            _logger.warn(_('On_demand content cannot be migrated without a remote '
-                                           'pulp2 unit_id: {}'.format(pulp2content.pulp2_id)))
-                        future_relations.update({'lces': list(pulp2lazycatalog)})
-                    else:
-                        relative_path = (
-                            pulp_2to3_detail_content.relative_path_for_content_artifact
-                        )
-                        da = DeclarativeArtifact(
-                            artifact=artifact,
-                            url=NOT_USED,
-                            relative_path=relative_path,
-                            remote=None,
-                            deferred_download=False)
-                        dc = DeclarativeContent(content=pulp3content, d_artifacts=[da])
-                        dc.extra_data = future_relations
-                        await self.put(dc)
-
                 if pb:
                     pb.increment()
 
                 if has_future and dc:
                     futures.append(dc)
-                resolve_futures = len(futures) >= DEFAULT_BATCH_SIZE or pb.done == pb.total
+                all_content_done = pb.done == pb.total or (pb.done + skipped_content) == pb.total
+                resolve_futures = len(futures) >= DEFAULT_BATCH_SIZE or all_content_done
```

Comment:
If you decide to have set future resolution outside of the loop as well, we can get rid off the second condition completely.